### PR TITLE
Warn when automatically lowering an inductive declared with `: Type` to Prop

### PR DESCRIPTION
**Pull request by SkySkimmer**
+ option to disable the lowering and option to default declare dependent eliminators for inductives with explicit Prop (for compat when fixing the warning by putting explicit Prop)

Inductives with no explicit type (eg `Inductive foo := C.`) can still get silently put in Prop.

Maybe it would make sense to support `_` as "pick Prop or Type as needed" to make it easier to port indexed inductives? eg `Inductive foo : nat -> Type := .` would become `Inductive foo : nat -> _ := .` (currently it fails saying "not an arity"). But it doesn't look that nice so probably not worth doing.

Funind behaviour is slightly changed as we now force its inductives in Type instead of letting them be lowered (forward compatible way to disable the warning).

Overlays:
- https://github.com/LPCIC/coq-elpi/pull/624

### CODE REVIEW

**Comment by SkySkimmer** [comment]
Maybe the options should be attributes instead, not sure.

File: doc/sphinx/language/core/inductive.rst [hunk 1]
```diff
@@ -148,6 +162,37 @@ by giving the type of its arguments alone.
...
+.. flag:: Automatic Proposition Inductives
+
+   By default the above behaviour is extended to empty and singleton
+   inductives explicitly declared in `Type` (but not those in explicit
+   universes using `Type@{u}`, or in `Type` through an auxiliary definition
+   such as `Definition typ := Type.`).
+
+   Disabling this flag prevents inductives with an explicit non-`Prop`
+   type from being lowered to `Prop`. This will become the default in
+   a future version. Note :flag:`Dependent Proposition Eliminators` to
```
Comment (jfehrle) [inline doc/sphinx/language/core/inductive.rst:182]:
```suggestion
   a future version. Use :flag:`Dependent Proposition Eliminators` to
```

File: doc/sphinx/language/core/inductive.rst [hunk 2]
```diff
@@ -148,6 +162,37 @@ by giving the type of its arguments alone.
 
       Inductive nat : Set := O | S (_:nat).
 
+Automatic Prop lowering
++++++++++++++++++++++++
+
+When an inductive is declared with no explicit sort, it is put in the
+smallest sort which does not prevent large elimination (excluding
```
Comment (jfehrle) [inline doc/sphinx/language/core/inductive.rst:169]:
```suggestion
When an inductive is declared without an explicit sort, it is put in the
smallest sort which permits large elimination (excluding
```

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Comment by ppedrot** [comment]
@coqbot run full ci

**Comment by SkySkimmer** [comment]
category theory failure is spurious

**Comment by ppedrot** [comment]
@coqbot run full ci

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**